### DISCUSSION TITLE
path traversal attack, Azure Web Application firewall rule

### DISCUSSION THREAD

**Discussion by spowell-216** [iTwin/itwinjs-core#3389] [Q&A]
We are testing a backend / frontend scenario in a private Azure cloud and our InfoSec team has enabled a web application firewall rule on our web application gateway that rejects traffic that has [path data](https://owasp.org/www-community/attacks/Path_Traversal) in the request .  This traffic seems to be primarily in the PresentationRpcInterface.  

![image](https://user-images.githubusercontent.com/51784707/159256340-048ccbc1-d7f0-4cdb-b09c-2b5b8c4bfa64.png)

we are still on version 2.19.28.  I know a lot of changes were made in version 3.x and reading through the change notes [presentation](https://www.itwinjs.org/changehistory/#presentation) nothing stood out with regards to this.  If we expedite our migration efforts to ver 3.x will this solve our problems or will we still have an issue where "path" data will be in the request payload?

**Comment by grigasp** [comment]
Thanks @spowell-216 for letting us know about the issue. It's not fixed in 3.0, but I added a work item to fix this ASAP. 

As a side note, I can confirm this path is not used anywhere on the backend and is only there because on the frontend the object is read from a JSON file that uses the `$schema` attribute to get intellisense.

**Comment by Alina657** [comment]
The issue is now fixed and will be available in release 3.2.
See [#3457](https://github.com/iTwin/itwinjs-core/pull/3457)
